src/OpenTelemetry/BatchExportProcessor.cs

```diff
@@ -268,7 +277,10 @@ private void ExporterProc()
 
                 if (this.circularBuffer.Count > 0)
                 {
-                    using (var batch = new Batch<T>(this.circularBuffer, this.maxExportBatchSize))
+                    using (var batch = new Batch<T>(this.circularBuffer, this.maxExportBatchSize)
+                    {
+                        CleanupAction = cleanupAction,
```

Comment:
What if there are multiple batch processors? Or another processor (e.g. simple) after a batch one?

Reply:
Tricky tricky! I added reference tracking to solve it. Check it out, LMK your thoughts.